### DISCUSSION TITLE
How does this tool work?

### DISCUSSION THREAD

**Discussion by Geek4Coding** [davetapley/helium-tax#46] [General]
I’m confused on how this tool works. Does it calculate how much each miner has made for a 24 hour period and is the price it’s showing in the calculations the price of hnt at the end of the 24 hr period?? Tax law states I need to pay taxes on the price hnt is at the time I received the rewards. Will irs care if it’s the market price at the end of the day or beginning even though the price was different for every earning?

**Comment by davetapley** [comment]
> Does it calculate how much each miner has made for a 24 hour period and is the price it’s showing in the calculations the price of hnt at the end of the 24 hr period?

No. The file you get will contain a separate line for each reward transaction (there are many of these per day), more info [here](https://docs.helium.com/blockchain/mining/).

---

> Tax law states I need to pay taxes on the price hnt is at the time I received the rewards
> Will irs care if it’s the market price at the end of the day or beginning even though the price was different for every earning?

Per [the docs](https://docs.helium.com/blockchain/oracles/):

> Every 10 blocks (roughly every 10 minutes) the blockchain will attempt to establish a new HNT/$USD price.

and this is the price used for each line.

I'm not a tax lawyer, so I can't make any guarantee whether this fits the IRS' requirement of "the fair market value", as given in Q.8 [here](https://www.irs.gov/irb/2014-16_IRB#NOT-2014-21).

**Reply by NeverHeardThat** [reply]
Thank you Dave for creating this awesome tool and making it available to everyone!  I've noticed that if you put in a wallet address to run a query the results appear to only be for the miner(s) in the wallet currently.  If for example, you had 2 miners in a wallet in 2020 but have since sold one, when you run the 2020 report by wallet address you will only get a list of transactions for the single miner still in the original wallet address today.

**Reply by davetapley** [reply]
Good catch @NeverHeardThat, thanks for reporting!
I've created a bug for this https://github.com/davetapley/helium-tax/issues/49, but it might take some time to get fixed because I think it'll need changes to the Helium API.
